transaction/src/action/position.rs

```diff
@@ -42,6 +42,22 @@ impl IsAction for PositionOpen {
 }
 
 impl PositionOpen {
+    /// Create a new valid `PositionOpen`.
+    pub fn new(position: Position, initial_reserves: Reserves) -> anyhow::Result<Self> {
+        if initial_reserves.r1.value() as u128 > MAX_RESERVE_AMOUNT
+            || initial_reserves.r2.value() as u128 > MAX_RESERVE_AMOUNT
+        {
+            Err(anyhow::anyhow!(format!(
+                "position reserves too big (limit: {MAX_RESERVE_AMOUNT})"
+            )))
+        } else {
+            Ok(Self {
+                position,
+                initial_reserves,
+            })
+        }
+    }
```

Comment:
Possibly counterintuitively, but I think it might be better to not have this constructor: either we decide that we want to make the excess-size values totally unrepresentable, or we should check them in the place where the check is required. Having a check in this constructor might give a false sense of security, because it's not the only way that a `PositionOpen` can be constructed.